mqtt_client.c

```diff
@@ -208,6 +209,10 @@ static esp_err_t esp_mqtt_set_ssl_transport_properties(esp_transport_list_handle
                      goto esp_mqtt_set_transport_failed);
     }
 
+    if (cfg->crt_bundle_attach != NULL) {
```

Comment:
This condition should be part of the above `if-else` block as these cert options are exclusive (either you set the certificate directly, or use the ca-store or the cert-bundle)